src/StudyEntities/Candidate/CandIDGenerator.php

```diff
@@ -35,6 +41,33 @@ public function generate(): CandID {
         );
     }
 
+    /**
+     * Wrapper for \Database::insert() to add new Candidate information to the
+     * database. This function will generate new IDs in the edge-case that a
+     * race condition occurs and causes an error upon insertion.
+     *
+     * @param array $setArray An associative array of column names and values.
+     *
+     * @return CandID The newly-created CandID object.
+     */
+    public function createIDAndInsertWithValues(array $setArray): CandID {
+        do {
+            $invalidID = false;
```

Comment:
The scoping of this is weird since it's used in the looping condition and I wouldn't be surprised if it eventually became a php/phan/phpcs/something warning about an undefined variable.

Reply:
Alrighty I'll move it